pkg/downloader/manager.go

```diff
@@ -509,6 +509,10 @@ func (m *Manager) ensureMissingRepos(repoNames map[string]string, deps []*chart.
 			continue
 		}
 
+		if _, ok := repoNames[dd.Repository]; ok {
+			continue
+		}
+
```

Comment:
`repoNames` has a structure of `[dd.Name] = dd.Repository`. So, this change isn't going to work. This section of code does need to be refactored to fix this bug. I see the bug.

Reply:
The repoNames is keyed on the chart name of the dependency, however that will not work in this case as there is more than one dependency with the same chart name.
I would suggest changing the key to the repo url, i.e. [dd.Repository] == repo.Name and refactoring methods 'resolveRepoNames', 'ensureMissingRepos' and 'resolve' accordingly

Reply:
@MichaelMorrisEst thanks so much.